On the fly variant mapping update

Fnyasimi opened the pull request:
The `--on_the_fly_variant_mapping`  wasn't working with variants whose alleles have been flipped, or are complements.

In this update I was able  to map those variants to be used in the prediction.

Fnyasimi commented:
@Heroico  kindly review this and provide some feedback

Heroico commented on software/metax/misc/Genomics.py:
```diff
@@ -5,16 +5,38 @@ def coordinate_format(format, chromosome, position, ref_allele, alt_allele):
+        c = format.format(chromosome, position, nt[ref_allele], nt[alt_allele])
+        # try swapped complement
+        c_ = format.format(chromosome, position, nt[alt_allele], nt[ref_allele])
+        if v_ in mapping:
```
This is confusing - it uses mapping as a mapping or as a filter based on type - which goes against the method name.
I would simply enforce mapping to be a properly built dictionary (asserting it's dict at the method beginning and removing the type check).

Also, the set seems to be initialized to `rsid` -why not use `varID` which actually contains the formatted ids in gtex models?

Fnyasimi replied:
The mapping is used as a filter in this case as it was implemented before when the mapping is a dict object. But since from the prior step I have my mapping as a set that is why I added additional type check. 

The set is initialized to `rsid` because the `rsid` is update to the value passed to `--model_db_snp_key`. If you pass in the `varID` to the argument then that is what will be used as the `rsid`.

Heroico replied:
Ok, thanks for the clarification.
To create a proper dict, I would load the `extra` table again with a dict where, for each row, `varID` column is the key and `rsid` is the value. Then the check for complements becomes more straightforward.

Heroico reviewed (commented):
What data was this tested on?

Fnyasimi commented on software/Predict.py:
```diff
@@ -113,6 +113,8 @@ def get_variant_mapping(args, weights):
         else:
             raise Exceptions.InvalidArguments("Unsupported variant mapping argument")
+    elif len(args.on_the_fly_mapping):
+        mapping = set(weights.rsid)
```
@Heroico thanks for the clarification. 

The reason I set the mapping to a set is because I wanted to utilize this [function](https://github.com/hakyimlab/MetaXcan/blob/master/software/Predict.py#L122) but again when we are using `--on_the_fly_mapping` I am not sure I we can get a dict like object to pass to that function so I opted to use the set and leading to the changes below. 

I would appreciate suggestion to handle this better.

Fnyasimi commented:
> 
> 
> What data was this tested on?

It was tested on data which I am currently working on. I also created a small debug db(20 variants with flipped alleles, complement alleles, and complement flipped alleles) and vcf (with the 20 variants and 20 individuals) for testing  purposes.

Heroico commented:
Hi @Fnyasimi , I suggest adding your test/debug data to the unit tests in this repo.

Fnyasimi commented:
@Heroico  I have modified the code not to use mapping and also added test data. The snps used should be 100% without using `--skip_palindromic` flag, when using the flag it should be 80%.

Heroico reviewed (approved):
Looks good,!

Fnyasimi commented:
@liangyy  you can merge this
